flycheck.el

```diff
@@ -5264,7 +5264,16 @@ function `buffer-file-name'."
       (when (seq-some (apply-partially #'flycheck-same-files-p
                                        (expand-file-name filename cwd))
                       buffer-files)
-        (setf (flycheck-error-filename err) (buffer-file-name)))))
+        (let ((new-filename (buffer-file-name)))
+          (setf (flycheck-error-filename err) new-filename
+                (flycheck-error-message err)
+                (replace-regexp-in-string
```

Comment:
Would this fit on the previous line? I had trouble parsing this visually at first :)